### ISSUE TITLE
Clear() and ResetToDefaults() have no implementation guide. 

### ISSUE THREAD

**Issue by bowlercaptain** [thesecretlab/YarnSpinner#47]
**TL;DR** It's not clear what VariableStorageBehaviour.Clear() and VariableStorageBehaviour.ResetToDefaults() are supposed to do.

I am currently integrating Yarnspinner into my game (thanks guys! So much is so much easier!), and I'm just about to finish the process, but while most has been well-explained or pretty self-evident (not too hard to decipher the design request in Value GetValue(string variableName)), I'm stuck on the meaning of ResetToDefaults() and Clear() in VariableStorageBehaviour, and how they differ. I can imagine that one of them is supposed to wipe the save file, but I believe the other gets called at the beginning of each playsession(?). I could not figure out from the code or docs what these functions are supposed to do. I would of course appreciate an explanation for myself, but more so I think this signals a gap in the docs. Maybe Clear() could get a more verbose, clearer (heh) name? Maybe those functions could get some comment-explanations like DialogueUIBehaviour has? I was about to recommend triple-slash documentation ([it's cool](https://msdn.microsoft.com/en-us/library/5ast78ax.aspx)! It shows up in tooltips in VS and can be really helpful!), but I notice that this does not appear when you're implementing an abstract class, so it can't be _quite_ as slick as one might hope.

**Comment by desplesda** [comment]
The documentation really does need a significant round of improvement, yeah. Thanks for pointing out this specific flaw - I'll make sure that it gets addressed!

I'll leave this issue open until I consider the documentation to be in a better situation.

**Comment by bowlercaptain** [comment]
Out of specific curiosity and perhaps to help you write docs later: What _are_ Clear and ResetToDefaults Supposed to do?

**Comment by PeteLawler** [comment]
@desplesda You have some comments on these calls, just wondering if, extending @bowlercaptain's thought above one is more intended for a new game with a blank profile and a new game with a default profile (as an eg)

**Comment by PeteLawler** [comment]
I've documented Clear as 'erase all variables' and ResetToDefaults as 'reset variables to default values'. I'm not entirely sure whether I can make this any more clear.

**Comment by TheMartianLife** [comment]
Given @PeteLawler has looked this over, this Issue seems closed. There are other issues open for general documentation improvement, which is definitely on our radar, else you are free to open another Issue if you have a particular concern :)
